stdlib/src/builtin/string.mojo

```diff
@@ -1666,49 +1681,22 @@ struct String(
         )
 
     fn isspace(self) -> Bool:
-        """Determines whether the given String is a python
-        whitespace String. This corresponds to Python's
+        """Determines whether every character in the given String is a
+        python whitespace String. This corresponds to Python's
         [universal separators](
             https://docs.python.org/3/library/stdtypes.html#str.splitlines)
-        `" \\t\\n\\r\\f\\v\\x1c\\x1e\\x85\\u2028\\u2029"`.
+        `" \\t\\n\\r\\f\\v\\x1c\\x1d\\x1e\\x85\\u2028\\u2029"`.
 
         Returns:
-            True if the String is one of the whitespace characters
+            True if the whole String is made up of whitespace characters
                 listed above, otherwise False.
         """
-        # TODO add line and paragraph separator as stringliteral
-        # once unicode escape secuences are accepted
-        var next_line = List[UInt8](0xC2, 0x85)
-        """TODO: \\x85"""
-        var unicode_line_sep = List[UInt8](0xE2, 0x80, 0xA8)
-        """TODO: \\u2028"""
-        var unicode_paragraph_sep = List[UInt8](0xE2, 0x80, 0xA9)
-        """TODO: \\u2029"""
-
-        @always_inline
-        fn _compare(
-            item1: UnsafePointer[UInt8], item2: UnsafePointer[UInt8], amnt: Int
-        ) -> Bool:
-            var ptr1 = DTypePointer(item1)
-            var ptr2 = DTypePointer(item2)
-            return memcmp(ptr1, ptr2, amnt) == 0
-
-        if len(self) == 0:
+
+        if self.byte_length() == 0:
             return False
 
         for s in self:
-            var no_null_len = len(s)
-            var ptr = s.unsafe_ptr()
-            if no_null_len == 1 and not _isspace(ptr[0]):
-                return False
-            elif no_null_len == 2 and not _compare(
-                ptr, next_line.unsafe_ptr(), 2
-            ):
-                return False
-            elif no_null_len == 3 and not (
-                _compare(ptr, unicode_line_sep.unsafe_ptr(), 3)
-                or _compare(ptr, unicode_paragraph_sep.unsafe_ptr(), 3)
-            ):
+            if not s.isspace():
```

Comment:
Given that you now have a `StringSlice.isspace()` implementation, why is a loop needed here? Could you just simply `return self.as_string_slice().isspace()` here?

Reply:
I think not since `StringSlice.isspace()` uses the length of the slice as a discriminant
```mojo
        @always_inline
        fn _compare(
            item1: UnsafePointer[UInt8], item2: UnsafePointer[UInt8], amnt: Int
        ) -> Bool:
            var ptr1 = DTypePointer(item1)
            var ptr2 = DTypePointer(item2)
            return memcmp(ptr1, ptr2, amnt) == 0

        var no_null_len = len(self)
        var ptr = self.unsafe_ptr()
        if no_null_len == 1 and _isspace(ptr[0]):
            return True
        elif no_null_len == 2 and _compare(ptr, next_line.unsafe_ptr(), 2):
            return True
        elif no_null_len == 3 and (
            _compare(ptr, unicode_line_sep.unsafe_ptr(), 3)
            or _compare(ptr, unicode_paragraph_sep.unsafe_ptr(), 3)
        ):
            return True
```